### PR TITLE
Langchain destination: Check pincone index dimensions as part of check

### PR DESCRIPTION
**Pull request by flash1293**
The check of this destination only validates the configured index is reachable, but a sync might still fail if the dimensions are not configured correctly. This PR makes this part of the check to have an actionable error message:
Screenshot 2023-08-02 at 16 52 21

### CODE REVIEW

**Comment by octavia-squidington-iii** [comment]
##  destination-langchain test report (commit [`a6b486d3bc`](https://github.com/airbytehq/airbyte/pull/28977/commits/a6b486d3bc8bc60f2637a8ba0ef14a7ffb949b67)) - ✅

⏲️  Total pipeline duration: 04mn40s 

| Step                                                                         | Result   |
|:-----------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-langchain/metadata.yaml | ✅        |
| Connector version semver check                                               | ✅        |
| QA checks                                                                    | ✅        |
| Code format checks                                                           | ✅        |
| Connector package install                                                    | ✅        |
| Build destination-langchain docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                   | ✅        |
| Integration tests                                                            | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci-internal/connectors/test/pull_request/flash1293_validate-index-dimensions/1690988142/a6b486d3bc8bc60f2637a8ba0ef14a7ffb949b67/destination-langchain/0.0.6/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-langchain test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-langchain test report (commit [`123caf1ac4`](https://github.com/airbytehq/airbyte/pull/28977/commits/123caf1ac4c9b00fb1ff979dcd753374393b4aef)) - ✅

⏲️  Total pipeline duration: 04mn12s 

| Step                                                                         | Result   |
|:-----------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-langchain/metadata.yaml | ✅        |
| Connector version semver check                                               | ✅        |
| QA checks                                                                    | ✅        |
| Code format checks                                                           | ✅        |
| Connector package install                                                    | ✅        |
| Build destination-langchain docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                   | ✅        |
| Integration tests                                                            | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci-internal/connectors/test/pull_request/flash1293_validate-index-dimensions/1690988550/123caf1ac4c9b00fb1ff979dcd753374393b4aef/destination-langchain/0.0.6/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-langchain test
```

File: airbyte-integrations/connectors/destination-langchain/destination_langchain/indexer.py [hunk 1]
```diff
@@ -95,7 +95,10 @@ def index(self, document_chunks, delete_ids):
 
     def check(self) -> Optional[str]:
         try:
-            pinecone.describe_index(self.config.index)
+            description = pinecone.describe_index(self.config.index)
+            actual_dimension = int(description.dimension)
+            if actual_dimension != self.embedder.embedding_dimensions:
+                return f"Your embedding configuration will produce vectors with dimension {self.embedder.embedding_dimensions:d}, but your index is configured with dimension {actual_dimension:d}. Make sure embedding and indexing configurations match."
```
Comment (aaronsteers) [inline airbyte-integrations/connectors/destination-langchain/destination_langchain/indexer.py:101]: 👍

**Review by aaronsteers — APPROVED** [review]
LGTM! 🚢

**Pull request merged**